atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java

```diff
@@ -719,11 +720,14 @@ private CompletableFuture<Channel> bootstrapClient(final Address address) {
     bootstrap.option(ChannelOption.TCP_NODELAY, true);
     bootstrap.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, 1000);
     bootstrap.group(clientGroup);
-    // TODO: Make this faster:
-    // http://normanmaurer.me/presentations/2014-facebook-eng-netty/slides.html#37.0
     bootstrap.channel(clientChannelClass);
-    bootstrap.remoteAddress(resolvedAddress, address.port());
+    bootstrap.resolver(
+        new DnsAddressResolverGroup(
+            new DnsNameResolverBuilder(clientGroup.next())
```

Comment:
:wrench: Please register a `LoggingDnsQueryLifeCycleObserverFactory` so that we can get DNS logs if needed